smtp/smtpc.c

```diff
@@ -440,6 +442,15 @@ smtpc_execute(struct smtpc_request *req, double timeout)
 		++env->stat.failed_requests;
 		smtpc_request_delete(req);
 		return -1;
+        case CURLE_SEND_ERROR:
+		{
+		char error_msg[CURL_ERROR_SIZE];
```

Comment:
> You must keep the associated buffer available until libcurl no longer needs it. Failing to do so will cause very odd behavior or even crashes. libcurl will need it until you call curl_easy_cleanup or you set the same option again to use a different pointer.

https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html

We don't call curl_easy_cleanup() before returning from this stack frame.

Reply:
We need the second buffer only to concatenate a constant string with given one, allocate a dynamic memory and copy the result to it? The second buffer looks redundant. Just allocate the dynamic memory and copy everything that is needed to it.